src/v/coproc/wasm_event_listener.cc

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright 2020 Vectorized, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/vectorizedio/redpanda/blob/master/licenses/rcl.md
+ */
+
+#include "coproc/wasm_event_listener.h"
+
+#include "config/configuration.h"
+#include "coproc/errc.h"
+#include "coproc/logger.h"
+#include "coproc/types.h"
+#include "coproc/wasm_event.h"
+#include "storage/directories.h"
+#include "utils/file_io.h"
+#include "utils/unresolved_address.h"
+#include "vassert.h"
+#include "vlog.h"
+
+#include <seastar/core/future.hh>
+#include <seastar/core/loop.hh>
+#include <seastar/core/seastar.hh>
+#include <seastar/core/sstring.hh>
+#include <seastar/net/socket_defs.hh>
+
+namespace coproc {
+
+using namespace std::chrono_literals;
+
+static wasm_event_action query_action(const iobuf& source_code) {
+    /// If this came from a remove event, the validator would
+    /// have failed if the value() field of the record wasn't
+    /// empty. Therefore checking if this iobuf is empty is a
+    /// certain way to know if the intended request was to
+    /// deploy or remove
+    return source_code.empty() ? wasm_event_action::remove
+                               : wasm_event_action::deploy;
+}
+
+/// Either writes or removes the file from disk. In either case, logs warning if
+/// the current state of the filesystem represents an unexpected issue
+ss::future<>
+wasm_event_listener::resolve_wasm_script(ss::sstring name, iobuf source_code) {
+    std::filesystem::path active_path(_active_dir / name.c_str());
+    return ss::file_exists(active_path.string())
```

Comment:
this could really benefit form a coroutine integration.

Reply:
@senior7515 should we give this a shot? or were you just making a comment? I agree